functions

```diff
@@ -28,11 +28,8 @@ emergency_shell() {
 }
 
 detect_virt() {
-   # Detect LXC containers
-   [ ! -e /proc/self/environ ] && return
-   if grep -q lxc /proc/self/environ >/dev/null; then
-       export VIRTUALIZATION=1
-   fi
+   # Detect LXC (and other) containers
+   [ "${container+x}" ] || export VIRTUALIZATION=1
```

Comment:
This makes no sense, it sets VIRTUALIZATION=1 when container is unset.

Use `[ -n "${container}" ] && export VIRTUALIZATION=1`

Reply:
See https://github.com/void-linux/void-runit/pull/9#discussion_r347925957, the code you're suggesting right now was the one I originally noted wasn't covering 100% of the cases (i.e. empty but set variable).

I'd suggest "just" changing the `||` to `&&`, however that'd yield a non-zero exit code for physical machines.
The current code looks like it would be possible to run it using `set -e`, 
Preferable would be either of these:

```bash
# personal favourite since it makes it obvious that `test` is a "regular" command albeit a builtin, and not a syntax construct (like the `[[ ]]` bashism or `(( ))` POSIX syntax.
test -z "${container+x}" || export VIRTUALIZATION=1

# using []
[ -z "${container+x}" ] || export VIRTUALIZATION=1

# shell-side negation with test
! test "${container+x}" || export VIRTUALIZATION=1

# shell-side negation with []
! test "${container+x}" || export VIRTUALIZATION=1
```

Reply:
alright I reversed the logic